### PR TITLE
Display a warning when nothing is searched

### PR DESCRIPTION
**Pull request by goto-engineering**
Relates to: https://github.com/BurntSushi/ripgrep/issues/1404

I brought back the "nothing searched message" when no files were matched.

Unsure how/where to add unit tests for this, I didn't find any tests for `main`.

### CODE REVIEW

File: crates/core/main.rs [hunk 1]
```diff
@@ -68,11 +68,20 @@ fn try_main(args: Args) -> Result<()> {
         process::exit(0)
     } else if messages::errored() {
         process::exit(2)
+    } else if !matched {
```
Comment (goto-engineering) [inline crates/core/main.rs:71]: I wasn't sure if the last `else` branch can be reached in any case that isn't the "nothing searched" case, so I made a new `else if`.

Comment (BurntSushi) [reply crates/core/main.rs:71]: Yes, the `else` branch is only reached when `matched` is false.

File: crates/core/main.rs [hunk 2]
```diff
@@ -68,11 +68,22 @@ fn try_main(args: Args) -> Result<()> {
...
+        eprint_nothing_searched();
+        process::exit(1)
```
Comment (BurntSushi) [inline crates/core/main.rs:73]: Hmmm, this is definitely not correct. :-( This is printing a message when nothing is _matched_, but what we want to do is print a message when nothing is _searched_.

**Review by BurntSushi — CHANGES_REQUESTED** [review]
Thanks! Unfortunately, this is not quite correct. I left a comment describing why below.

> Unsure how/where to add unit tests for this, I didn't find any tests for main.

`main` is probably the most tested routine in all of ripgrep. Tests for this probably want to go in `tests/feature.rs`.

File: crates/core/main.rs (hunk 2 above)
Comment (goto-engineering) [reply crates/core/main.rs:73]: Oh, I see. I can't seem to find the logic where the error message used to be shown, i.e. where we can conclude that there are no search candidates. Do you happen to know the area so I can hone in on it? I'm mostly looking in `walk.rs` right now.

Comment (BurntSushi) [reply crates/core/main.rs:73]: I would keep trying in `main.rs`. In particular, I believe the logic will need to be in both the `search` and `search_parallel` routines. For the former, I think it needs to check if `searcher.search()` is ever executed. Similarly for the latter.

**Pull request closed without merging**